src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -103,6 +104,10 @@ protected function processValue(mixed $value, bool $isRoot = false): mixed
 
     private function doProcessValue(mixed $value, bool $isRoot = false): mixed
     {
+        if ($value instanceof TypedReference) {
+            $value = $this->processAttributes($value);
```

Comment:
shouldn't this be merged with the next "if"? do we want to support returning a TypedRef from processAttributes?

Reply:
I thought about this but since `processAttributes()` might not return a `TypedReference` the rest of the if block would be invalid. Or am I missing something?